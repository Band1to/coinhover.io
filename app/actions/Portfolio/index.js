import * as R from 'ramda';
import * as api from '../../services/api';
import { linkLogo, formatPriceUSD, zeroBalanceValue } from '../../utils/modifier';

import {
  ADD_COIN_PORTFOLIO,
  UPDATE_COIN_BALANCE,
  REMOVE_COIN_PORTFOLIO
} from '../../actionTypes';

// export const ADD_COIN = 'ADD_COIN'
// export const REMOVE_COIN = 'REMOVE_COIN'
// export const UPDATE_COIN_VALUE = 'UPDATE_COIN_VALUE'

// action creators /////////////////////////////////////////////////////////////
export function add(portfolio) {
  console.log('actions add', portfolio);
  return {
    type: ADD_COIN_PORTFOLIO,
    portfolio
  };
}

export function remove(coin) {
  console.log('actions remove', coin);
  return {
    type: REMOVE_COIN_PORTFOLIO,
    coin
  };
}

export function update(coin) {
  console.log('actions update', coin);
  return {
    type: UPDATE_COIN_BALANCE,
    coin
  };
}

// actions /////////////////////////////////////////////////////////////////////
export function addCoin(coin) {
  return dispatch =>
    api.getCoin(coin.id)
      .then(res => R.head(res.data))
      .then(remoteCoin => linkLogo(coin, remoteCoin))
      .then(remoteCoin => zeroBalanceValue(remoteCoin))
      .then(remoteCoin => formatPriceUSD(remoteCoin))
      .then(remoteCoin => dispatch(add(remoteCoin)));
}

export function removeCoin(coin) {
  return dispatch => dispatch(remove(coin));
}

export function updateCoin(coin) {
  return dispatch => dispatch(update(coin));
}
