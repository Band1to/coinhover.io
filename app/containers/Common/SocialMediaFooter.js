import React from 'react';

export default function SocialMediaFooter() {
  return (
    <footer>
      <ul>
        <li>
          <em>&copy; { new Date().getFullYear() } coinhover.io</em>
        </li>
        <li>
          <a href="https://github.com/CoinHover" target="_blank" title="Follow on Github">
            <div className="icon-github" />
          </a>
        </li>
        <li>
          <a href="https://twitter.com/coinhover" target="_blank" title="Follow on Twitter">
            <div className="icon-twitter" />
          </a>
        </li>
        <li>
          <a href="https://www.facebook.com/coinhover/" target="_blank" title="Follow on Facebook">
            <div className="icon-facebook-squared" />
          </a>
        </li>
      </ul>
    </footer>
  );
}
