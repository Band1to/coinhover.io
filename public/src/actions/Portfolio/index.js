import * as R from 'ramda'
import * as api from '../../services/api'
import { linkLogo, formatPriceUSD, zeroBalanceValue } from '../../utils/modifier'

import * as actionTypes from '../../actionTypes'

// export const ADD_COIN = 'ADD_COIN'
// export const REMOVE_COIN = 'REMOVE_COIN'
// export const UPDATE_COIN_VALUE = 'UPDATE_COIN_VALUE'

// actions /////////////////////////////////////////////////////////////////////
export function addCoin(coin) {
	return dispatch =>
		api.getCoin(coin.id)
			.then((res) => R.head(res.data))
			.then((remote_coin) => linkLogo(coin, remote_coin))
			.then((remote_coin) => zeroBalanceValue(remote_coin))
			.then((remote_coin) => formatPriceUSD(remote_coin))
			.then((remote_coin) => dispatch(add(remote_coin)));
}

export function removeCoin(coin) {
	return dispatch => dispatch(remove(coin));
}

export function updateCoin(coin) {
	return dispatch => dispatch(update(coin));
}

// action creators /////////////////////////////////////////////////////////////
export function add(portfolio) {
	return {
		type: actionTypes.ADD_COIN_PORTFOLIO,
		portfolio
	}
}

export function remove(coin) {
	return {
		type: actionTypes.REMOVE_COIN_PORTFOLIO,
		coin
	}
}

export function update(coin) {
	return {
		type: UPDATE_COIN_VALUE,
		coin
	}
}
