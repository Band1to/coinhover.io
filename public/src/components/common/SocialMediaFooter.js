import React from 'react'

export default function SocialMediaFooter() {
	return (
		<footer>
			<ul>
				<li>
					<em>&copy; { new Date().getFullYear() } coinhover.io</em>
				</li>
				<li>
					<a href="https://github.com/CoinHover" target="_blank" title="Follow on Github">
						<div className="icon-github"></div>
					</a>
				</li>
				<li>
					<a href="https://twitter.com/coinhover" target="_blank" title="Follow on Twitter">
						<div className="icon-twitter"></div>
					</a>
				</li>
				<li>
					<a href="https://www.facebook.com/coinhover/" target="_blank" title="Follow on Facebook">
						<div className="icon-facebook-squared"></div>
					</a>
				</li>
			</ul>
		</footer>
	)
}