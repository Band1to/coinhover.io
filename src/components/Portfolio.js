import React from 'react'
import PropTypes from 'prop-types'

export default class Portfolio extends React.Component {
	constructor(props) {
		super(props)
		this.state = {

		}
	}

	render() {
		const addCoin = () => {
			console.log('addCoin...');
		};

		return (
			<div className="app-bg">
				<section className="welcome">
					<header>					
						<h1>CoinHover</h1>
						<h2>Watch all your balances</h2>
					</header>
					<section className="coin-table" onClick={ addCoin }>
						<p>Add Coin</p>
						<div className="ico-add"></div>
						<div className="coin-input"></div>
					</section>
				</section>
			</div>
		)
	}
}