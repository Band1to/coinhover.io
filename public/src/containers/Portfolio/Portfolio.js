import React from 'react';
import Header from '../common/Header'
import AssetsTable from '../assetsTable/AssetsTable'
import SocialMediaFooter from '../common/SocialMediaFooter'
import * as api from '../../services/api'
// import local_coins from '../../coins.json'
// import { updateLocalCoins } from '../../services/coinFactory'
// let localCoins = local_coins;

export default class Portfolio extends React.Component {
	constructor(props) {
		super(props)
		this.state = {
			loading: true,
		};
	}

	componentDidMount() {
		// console.log('Portfolio state', this.state);
		// api.getAllCoins().then((res) => {
		// 	const portfolioCoins = updateLocalCoins(localCoins, res.data);
		// 	updatePortfolio(portfolioCoins);
		// 	this.setState({ assets: portfolioCoins, loading: false });
		// });
		this.setState({ loading: false });
	}

	render() {
		return (
			<div className="app-bg">
				<section className="portfolio">
					<Header />
					{ this.state.loading ? (
						<div className="loading">
							<div className="loader"></div>
							<span>Loading coin data...</span>
						</div>
					) : (
						<AssetsTable />
					)}
					<SocialMediaFooter />
				</section>
			</div>
		)
	}
}
