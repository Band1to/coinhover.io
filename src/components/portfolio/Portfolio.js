import React from 'react'
import { connect } from 'react-redux'
import SocialMediaFooter from '../common/SocialMediaFooter'
import AssetsTable from '../assetsTable/AssetsTable'
// import { assets } from '../../models/temp'
import local_coins from '../../coins.json'
import * as api from '../../services/api'
import { updateLocalCoins } from '../../services/coinFactory'

const mapStateToProps = ({ portfolio }) => ({
	portfolio
});

let localCoins = local_coins;

class Portfolio extends React.Component {
	constructor(props) {
		super(props)
		this.state = {
			loading: true,
			assets: props.portfolio,
			total: 0
		};
		console.log('props', props);
	}

	componentDidMount() {
		// api.getAllCoins().then((res) => {
		// 	const portfolioCoins = updateLocalCoins(localCoins, res.data);
		// 	updatePortfolio(portfolioCoins);
		// 	this.setState({ assets: portfolioCoins, loading: false });
		// });

		// this.setState({ assets: local_coins, loading: false });
		// this.setState({ assets: [], loading: false });
		this.setState({ loading: false });
	}

	render() {
		const assets = this.state.assets;
		const total  = this.state.total;

		return (
			<div className="app-bg">
				<section className="portfolio">
					<header>
						<h1><span className="plus">+</span>COINHOVER</h1>
						<h2>Watch your cryptocurrency asset balances in once place.</h2>
						<em className="num">${ total }</em>
					</header>
					{ this.state.loading ? (
						<div className="loading">
							<div className="loader"></div>
							<span>Loading coin data...</span>
						</div>
					) : (
						<AssetsTable assets={ assets }/>
					)}
					<SocialMediaFooter />
				</section>
			</div>
		)
	}
}

export default connect(mapStateToProps, null)(Portfolio)