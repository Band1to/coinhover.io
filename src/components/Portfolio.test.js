import React from 'react'
import * as enzyme from 'enzyme';
import toJson from 'enzyme-to-json'
import Portfolio from './Portfolio'
import SocialMediaFooter from './common/SocialMediaFooter'

const portfolio = enzyme.shallow(<Portfolio />);

describe('<Portfolio /> component', () => {
	it('should render', () => {
		const tree = toJson(portfolio);
		expect(tree).toMatchSnapshot();
	});

	it('header contains the correct text', () => {
		expect(portfolio.find(".welcome h1").text()).toEqual('COINHOVER');
		expect(portfolio.find(".welcome h2").text()).toEqual('A cryptocurrency asset balance calculator');
	});

	it('contains the SocialMediaFooter component', () => {
		expect(portfolio.find(SocialMediaFooter).length).toBe(1);
	});
});