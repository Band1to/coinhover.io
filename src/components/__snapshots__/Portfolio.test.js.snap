// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Portfolio /> component should render 1`] = `
<div
  className="app-bg"
>
  <section
    className="welcome"
  >
    <header>
      <h1>
        COINHOVER
      </h1>
      <h2>
        A cryptocurrency asset balance calculator
      </h2>
      <p>
        Currently in development...
      </p>
    </header>
    <SocialMediaFooter />
  </section>
</div>
`;
